awsrds/rds_db_instance.go

```diff
@@ -593,6 +605,13 @@ func (r *RDSDBInstance) GetFullValidTargetVersion(engine string, currentVersion
 	}
 
 	if len(engineVersionsOut.DBEngineVersions[0].ValidUpgradeTarget) == 0 {
+		if currentSemVersion.Major() == targetMonikerSemVer.Major() {
+			logSess.Info("no-new-version-but-same-major-noop", lager.Data{
+				"target-version-moniker": targetVersionMoniker,
+				"current-version": currentVersion,
+			})
+			return "", nil
```

Comment:
We should mention in the function's doc/comment that this return value is possible and what it means.

Reply:
I've added a comment about this.